libraries/api-extractor-model/src/mixins/ApiItemContainerMixin.ts

```diff
@@ -134,7 +137,7 @@ export function ApiItemContainerMixin<TBaseClass extends IApiItemConstructor>(
       jsonObject: IApiItemContainerJson
     ): void {
       baseClass.onDeserializeInto(options, context, jsonObject);
-
+      options.shouldSortMembers = false;
```

Comment:
If I understand correctly, this line will always disable sorting when loading an already serialized .api.json file.  I tis a change in behavior -- prior to this PR, if the `ApiItem.getSortKey()` rules were adjusted, then the new sort order would be applied when loading old .api.json files.  

I don't care too much about whether old files get resorted, however that decision characterizes this API:

1. **api-extractor decides:**  If the .api.json file is the final answer about order, then `ApiItem.getContainerKey()` is not really part of the data model -- it is more like a helper that API Extractor may optionally use when sorting containers.  There is no need to store `shouldSortMembers` in the data file. (For backwards compatibility, `shouldSortMembers` should be true if-and-only-if the .api.json file is an old version.) If we go those route, then we need some docs to make it very obvious in what situations `ApiItem.getContainerKey()` will be ignored.  **- OR -**

2. **api-extractor-model decides:**  On the other hand, suppose that `api-extractor-model` implements sorting normally. And if the sort rule is improved in a new version, then old .api.json files will get re-sorted when they are loaded.  In this design, the `shouldSortMembers` does need to be serialized into the .api.json file so that the loader knows whether to sort or not.

In the current PR, we have a sort of hybrid of these two designs, where `api-extractor-model` decides during class construction but not deserialization, which I think is a little counterintuitive (given that the deserializer calls the class constructor).

Here's one other possibility:

3. **we always sort like before:**  We could add a `originalOrder: number | undefined` property to `ApiEnumMember`, and if this is not undefined then `getContainerKey()` will sort using this number instead of the name. Then API Extractor can assign this number to produce an arbitrary ordering. The downside of this approach is that `order` would need to get serialized into the .api.json file format.

Since this PR has been pending for 4 weeks, I can volunteer to do the work. 😋  Just let me know which way you think we should go.

Reply:
I chatted with @zelliott about this offline. We've decided to go with option 2.  Since our convention is for boolean options to default to `false`, I'm proposing to rename `shouldSortMembers` -> `preserveMemberOrder`. (We also considered using `EnumMemberOrder` directly, but `ApiItemContainerMixin`  is too abstract to implement a choice like `"by-value"` -- instead I think the subclass would customize the `getContainerKey()` rather than disabling sorting.)

Reply:
Proposed solution: https://github.com/mrshllstock/rushstack/pull/1

Reply:
If you guys are good with the changes in my PR above, then I think we're otherwise good to merge this feature. ✔